src/data/finsupp/witnesses.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.finsupp.basic
+
+/-!
+#  `diff` for finitely supported functions
+
+Let `α N` be two Types, and assume that `N` has a `0` and let `f g : α →₀ N` be finitely supported
+functions.
+
+##  Main definition
+
+*  `finsupp.diff f g : finset α`, the finite subset of `α` where `f` and `g` differ,
+
+In the case in which `N` is an additive group, `finsupp.diff f g` coincides with
+`finsupp.support (f - g)`.
+-/
+
+variables {α N : Type*}
+
+namespace finsupp
+variables [decidable_eq α] [decidable_eq N]
+
+section diff
+
+section N_has_zero
+variables [has_zero N] {f g : α →₀ N}
+
+/--  Given two finitely supported functions `f g : α →₀ N`, `finsupp.diff f g` is the `finset`
+where `f` and `g` differ. This generalizes `(f - g).support` to situations without subtraction. -/
+def diff (f g : α →₀ N) : finset α :=
```

Comment:
One other possibility would be to call this `ne_locus` to match `ring_hom.eq_locus` and similarly named definitions. Let's see if any other reviewers have opinions

Reply:
I like `ne_locus`.  I used `diff` since `diff` is somewhat related to the symmetric difference of the supports (and `diff` contains the symmetric difference).

Reply:
A very small advantage of `diff` over `ne_locus` is that it is shorter.

Reply:
Once the name settles, I will also rename the file to match the chosen name: `diff` or `ne_locus` (or whatever else).